### ISSUE TITLE
Ready for 3.6.0 full PyPI/GitHub release

### ISSUE THREAD

**Issue by sbillinge** [diffpy/diffpy.utils#309]
### PyPI/GitHub full-release preparation checklist:

- [x] Create a new conda environment and install the rc from PyPI (`pip install <package-name>==??`)
- [x] License information on PyPI is correct.
- [x] Docs are deployed successfully to `https://www.diffpy.org/<package-name>`.
- [x] Successfully run all tests, tutorial examples or do functional testing.

Please let @sbillinge know that all checks are done and the package is ready for full release.

### conda-forge release preparation checklist:

- [x] Ensure that the full release has appeared on PyPI successfully.
- [x] New package dependencies listed in `conda.txt` and `test.txt` are added to `meta.yaml` in the feedstock.
- [x] Close any open issues on the feedstock. Reach out to @bobleesj if you have questions.
- [x] Tag @sbillinge and @bobleesj for conda-forge release.

### Post-release checklist

- [x] Run tutorial examples and conduct functional testing using the installation guide in the README. Attach screenshots/results as comments.
- [x] Documentation (README, tutorials, API references, and websites) is deployed without broken links or missing figures.

**Comment by yucongalicechen** [comment]
The version shows up correctly and functions work.
![image](https://github.com/user-attachments/assets/60404d91-a159-4f07-a969-98817f7f6f2e)

**Comment by sbillinge** [comment]
I did a `3.6.0rc1` release to try and trigger the doc building workflow, but for some reason I don't think it ran? @bobleesj ?

**Comment by bobleesj** [comment]
Yes, replied here https://github.com/diffpy/diffpy.utils/issues/308#issuecomment-2566759122

but prob it's better to communicate here moving forward

**Comment by sbillinge** [comment]
@yucongalicechen NO need to build an env and test rc1 or higher.....this was just to try and test the pre-release doc bulid.  But we do want to build and look at the docs before doing the full release...

**Comment by sbillinge** [comment]
ok, I pushed rc2.   @bobleesj , how do we access the docs when they are deployed?  Do they just deploy to the usual github.io place or is there a special rc place?

**Comment by bobleesj** [comment]
> Do they just deploy to the usual github.io place or is there a special rc place?

Yes, the current setup just deployed to the main URL.  diffpy.github.io/diffpy.utils

> I pushed rc2

Dang it, the GitHub doc workflow CI isn't triggering on rc2.

@sbillinge 

I will need to fix and test this with a dummy org/repo. Dinner time soon... maybe continue tmr morning or sth? hopefully you can push a tag or merge while you are in your train?

**Comment by bobleesj** [comment]
> Do they just deploy to the usual github.io place or is there a special rc place?

Maybe we need a better instruction on this in the checklist. Should we later look into deploying a separate URL for rc versions? Maybe? For some packages (pdffit2...) may have a delay between rc release and full release.

**Comment by sbillinge** [comment]
> > Do they just deploy to the usual github.io place or is there a special rc place?
> 
> Maybe we need a better instruction on this in the checklist. Should we later look into deploying a separate URL for rc versions? Maybe? For some packages (pdffit2...) may have a delay between rc release and full release.

No, let's keep it simple.  If there is a big boo boo and we want to revert the docs we can just rerun the previous full release version with workflow_dispatch or sthg.

**Comment by sbillinge** [comment]
> > Do they just deploy to the usual github.io place or is there a special rc place?
> 
> Yes, the current setup just deployed to the main URL. diffpy.github.io/diffpy.utils
> 
> > I pushed rc2
> 
> Dang it, the GitHub doc workflow CI isn't triggering on rc2.
> 
> @sbillinge
> 
> I will need to fix and test this with a dummy org/repo. Dinner time soon... maybe continue tmr morning or sth? hopefully you can push a tag or merge on the train?

OK, sounds good.  Signing off for tonight.  I should have enough bandwidth tomorrow to do it.

**Comment by bobleesj** [comment]
@yucongalicechen a quick Q. Do you have access to check off items in the issue checklist? I am asking since this issue is created by @sbillinge

**Comment by bobleesj** [comment]
@sbillinge saw `3.6.0rc3` released 

<img width="1017" alt="Screenshot 2025-01-01 at 2 05 30 PM" src="https://github.com/user-attachments/assets/5c1f3ce2-48af-4c67-bcea-19048ce41e1a" />

Seems all good with doc showing the latest github tag as expected

<img width="767" alt="Screenshot 2025-01-01 at 2 05 47 PM" src="https://github.com/user-attachments/assets/493f592b-d6e4-4cf7-a4c1-574eb16e9c79" />

**Comment by bobleesj** [comment]
@sbillinge Time for `3.6.0` full release. Let's see if we have `Release Notes` displaying 3.6.0 and 3.6.0 tag commit equal to that of `main`..

**Comment by sbillinge** [comment]
I will put a shift in on the docs now before we do the full release.  Give me an hour or so.

**Comment by bobleesj** [comment]
noted

**Comment by yucongalicechen** [comment]
> @yucongalicechen a quick Q. Do you have access to check off items in the issue checklist? I am asking since this issue is created by @sbillinge

Nope!

**Comment by sbillinge** [comment]
just let me know in comments and I can check htem.  We can handle this differently next time.

**Comment by sbillinge** [comment]
@bobleesj @yucongalicechen I push a pr on the docs.  I will merge it and try another rc release to see what it looks like built.  If it looks good, I think we are ady fo full release if @yucongalicechen can sign off on everything.

**Comment by yucongalicechen** [comment]
docs look good to me

**Comment by sbillinge** [comment]
ok, we are go for launch.  I will do the full release now.

**Comment by bobleesj** [comment]
OK

**Comment by bobleesj** [comment]
Good, tag delete and creation work. 

But, I see something must have been broken with news items:

Our release note is fine: https://github.com/diffpy/diffpy.utils/releases/tag/3.6.0
<img width="929" alt="Screenshot 2025-01-01 at 3 48 34 PM" src="https://github.com/user-attachments/assets/03436637-f00d-4156-8321-a01c0923fd20" />

But our release note from CHANGELOG.rst must have some typos, etc. 

<img width="838" alt="Screenshot 2025-01-01 at 3 47 39 PM" src="https://github.com/user-attachments/assets/c981c4c5-59ce-459a-b494-6d404c8c7f20" />

I can make a PR to `main` and maybe we could render doc again by workflow dispatch?

**Comment by sbillinge** [comment]
quick investigation I notice that this may correlate with when there is a line-feed in the readme.  This may upset the `rst.`

**Comment by sbillinge** [comment]
Anyway, fantastic that all the workflows are working now! yipeeee!

**Comment by bobleesj** [comment]
It's like a mini rocket launch at the push of a button (tag)..

**Comment by sbillinge** [comment]
Can we do the conda-forge workflow now?  I want to send a slack to the group and have everyone update.

**Comment by sbillinge** [comment]
@bobleesj , you can go ahead with CF release once @yucongalicechen signs off.

**Comment by bobleesj** [comment]
One of the news (a line break):

<img width="876" alt="Screenshot 2025-01-01 at 3 58 36 PM" src="https://github.com/user-attachments/assets/cef4316c-4add-4885-a4c3-9121295d4104" />

and our changelog.rst:

<img width="945" alt="Screenshot 2025-01-01 at 3 59 06 PM" src="https://github.com/user-attachments/assets/080d29f7-6c37-406a-b708-4c3a0908076d" />

Ah, yeah .rst can't handle line-break. I will make sure to write that in our cookiecutter doc.

**Comment by bobleesj** [comment]
> @bobleesj , you can go ahead with CF release once @yucongalicechen signs off.

Yup I will

**Comment by yucongalicechen** [comment]
I'll update the PR on feedstock

**Comment by yucongalicechen** [comment]
Sorry just one last question - shall we expect the newly added packages (e.g. scipy, xraydb) to appear in `meta.yml` (this is a screenshot of that file)?
![image](https://github.com/user-attachments/assets/97e20de8-a7c3-4f9c-9c90-4bdee69c7bd7)

**Comment by bobleesj** [comment]
@yucongalicechen meta.yaml is managed manually so if you've added new dependencies, ensure that you also add under `run:` as well as update `requirements/conda.txt`

**Comment by bobleesj** [comment]
<img width="578" alt="Screenshot 2025-01-01 at 4 14 13 PM" src="https://github.com/user-attachments/assets/f9c7f45a-e824-44f9-859a-342a8e7592eb" />

For a bit more context on `conda.txt`

**Comment by sbillinge** [comment]
also make sure tests dependencies in meta.yml match `requirements/test.txt`

**Comment by yucongalicechen** [comment]
Gotcha, thanks! Adding them now.

**Comment by sbillinge** [comment]
@bobleesj I merged into CF

**Comment by sbillinge** [comment]
@yucongalicechen please can you do the last bullet above and we can close this.

**Comment by sbillinge** [comment]
ugh, @bobleesj deja vu (but I don't remember what the fix was)
![image](https://github.com/user-attachments/assets/9267983c-642a-4dfe-84b2-eeabacf4e689)

**Comment by bobleesj** [comment]
Yeah.. 0.0.1 (default when tag not found) cuz 'main' was used for doc deployment and it fails to get the tag associated with 'main' 

Hm. We need to ensure that when we deploy by dispatch from 'main', we also get the latest tag info. This will need another round of investigation...

**Comment by sbillinge** [comment]
gotcha.

**Comment by sbillinge** [comment]
@bobleesj I fixed it manually for now.

**Comment by yucongalicechen** [comment]
@sbillinge I was away for a sec sorry. Anything else I need to do?

**Comment by bobleesj** [comment]
So could you check the last bit of gitlab release instruction? It would instruct you to close this issue @yucongalicechen

**Comment by bobleesj** [comment]
> @bobleesj I fixed it manually for now.

Good idea.. on my phone and checked the fix on the web. Will be away from computer and will report back later tmr

**Comment by yucongalicechen** [comment]
Okay, I ran a couple examples from tutorial. They're working nicely. I think we can close this issue @sbillinge 
![image1](https://github.com/user-attachments/assets/f758505a-9722-4bb9-9cdf-dda8a248c09d)
![image2](https://github.com/user-attachments/assets/b843027d-00a3-40a4-8cf3-b02d18a6f060)

**Comment by sbillinge** [comment]
closed as completed

**Comment by bobleesj** [comment]
Happy new year! A great way to start 2025

**Comment by yucongalicechen** [comment]
Happy new year!! Thanks for all the comments and suggestions - I'm learning a lot during the process 😊!!

**Comment by sbillinge** [comment]
indeed, same to you guys.  And thanks for your efforts!  Huge deliverable!
